opossum-tool/OpossumUI
# feat: offer project search suggestions

### Summary of changes

- offer suggestions for projects (github, gitlab, bitbucket) when user searches by package name
- immediately get URL and license when user selects a project via the plus button
- include both name and namespace when searching by package name
- extend list of common package types
- remove golang from list of types that are requiring namespaces because there is debate in the community about this
- do not include preferred attributions in autocomplete suggestions because they are meant for a specific resource-attribution combination
- add optional group action to autocomplete component

![image](https://github.com/opossum-tool/OpossumUI/assets/46091730/ef92510b-2679-494a-821a-6684c8e3076d)

### Context and reason for change

Closes #2396 

### How can the changes be tested

Test all possible systems for which we offer suggestions. Here a list with search terms that will have at least one suggestion for the given system.

PROJECTS:
- github: "radash"
- bitbucket: "kde"
- gitlab: "gitlab-runner"

Note that repo URL and license are filled in immediately when clicking "+".
There are never any version suggestions for projects.

PACKAGES:
- npm: "react" or "lodash"
- maven: "jgiven" or "spring"
- golang: "authboss"
- pypi: "numpy"
- nuget: "serilog"
- cargo: "tokio"

Note that repo URL and license are filled in only after you choose a version in a second step after choosing a name.
